api/src/main/scala/Geoprocessing.scala

```diff
@@ -13,917 +13,45 @@ import geotrellis.vector._
-      "List(90, 6)" -> 3490,
-      "List(90, 60)" -> 8,
-      "List(90, 62)" -> 8,
-      "List(90, 64)" -> 6,
-      "List(90, 67)" -> 5,
-      "List(90, 70)" -> 9,
-      "List(90, 72)" -> 3,
-      "List(90, 75)" -> 6,
-      "List(90, 78)" -> 7,
-      "List(90, 8)" -> 1980,
-      "List(90, 80)" -> 4,
-      "List(90, 83)" -> 5,
-      "List(90, 86)" -> 1,
-      "List(90, 90)" -> 2,
-      "List(90, 93)" -> 1,
-      "List(95, 0)" -> 1502,
-      "List(95, 1)" -> 1133,
-      "List(95, 10)" -> 114,
-      "List(95, 12)" -> 89,
-      "List(95, 14)" -> 77,
-      "List(95, 15)" -> 42,
-      "List(95, 17)" -> 29,
-      "List(95, 19)" -> 32,
-      "List(95, 21)" -> 23,
-      "List(95, 23)" -> 17,
-      "List(95, 24)" -> 8,
-      "List(95, 26)" -> 12,
-      "List(95, 28)" -> 9,
-      "List(95, 3)" -> 597,
-      "List(95, 30)" -> 7,
-      "List(95, 32)" -> 5,
-      "List(95, 34)" -> 4,
-      "List(95, 36)" -> 4,
-      "List(95, 38)" -> 6,
-      "List(95, 40)" -> 3,
-      "List(95, 42)" -> 2,
-      "List(95, 44)" -> 1,
-      "List(95, 46)" -> 1,
-      "List(95, 48)" -> 4,
-      "List(95, 5)" -> 430,
-      "List(95, 6)" -> 280,
-      "List(95, 8)" -> 196
-    ),
-    "slope" -> Map ("List(0)" -> 50.38815789473684),
-    "nlcd_kfactor" -> Map (
-      "List(0)" -> 0.20602414696397067,
-      "List(11)" -> 0.05508165069169296,
-      "List(21)" -> 0.21868470419874417,
-      "List(22)" -> 0.16695226874641336,
-      "List(23)" -> 0.112148292194999,
-      "List(24)" -> 0.06672464966401591,
-      "List(31)" -> 0.07092705045562692,
-      "List(41)" -> 0.18912035761808532,
-      "List(42)" -> 0.206511688111982,
-      "List(43)" -> 0.19906280961858158,
-      "List(52)" -> 0.2698116178955606,
-      "List(71)" -> 0.22942145138570946,
-      "List(81)" -> 0.23793733590310687,
-      "List(82)" -> 0.2559927368472959,
-      "List(90)" -> 0.31187355224088653,
-      "List(95)" -> 0.24373768152729858
-    )
-  )
 
-  def getMultiOperations(input: MultiInput): Map[String, Map[String, Map[String, Double]]] =
-    // TODO Replace mock response with actual code
-    input.shapes.map(huc => huc.id -> input.operations.map(op =>
-      op.label -> mocks(op.label)
-    ).toMap).toMap
+  @throws(classOf[MissingStreamLinesException])
+  @throws(classOf[MissingTargetRasterException])
+  def getMultiOperations(input: MultiInput): Map[String, Map[String, Map[String, Double]]] = {
+    val hucs = input.shapes.map(huc => huc.id -> normalizeHuc(huc))
+    val rasters = collectRastersForInput(input, hucs.map(_._2))
+    val cachedOpts = input.operations.map(op => op -> getRasterizerOptions(op.pixelIsArea)).toMap
+
+    hucs.map(huc => huc._1 -> input.operations.map(op => {
+      val shape = huc._2
+      val opts = cachedOpts(op)
+      val layers = op.rasters.map(r => rasters(r))
+      val result = op.name match {
+        case "RasterGroupedCount" =>
+          rasterGroupedCount(layers, shape, opts).mapValues(_.toDouble)
```

Comment:
This and `rasterLinesJoin` below return `Map[String, Int]` but to match the type signature of this method I have to convert them to `Double`. Is there a type signature we can use that doesn't require this?